### ISSUE TITLE
Client triggered an unexpected error [Neo.DatabaseError.General.UnknownError]: head of empty list, reference 8fbc6196-5d99-41e5-98d7-524da14cc84e

### ISSUE THREAD

**Issue by gkostkowski** [neo4j/neo4j#12146]
I have constantly received logs about error "head of empty list" every few days. Usually it occurs during executing time-consuming query, but I cannot exactly point out what it may cause it, because it happens during various types of operations - usually it's during adding something to database, sometimes during query which alter changes some data or just selects to present. As a result, performance of dbms becomes very low or even current operation gets blocked. Restarting DBMS temporarily fixes issue.  
I assume that it's rather generic error and know that without reproduction steps it may be hard to spot the problem but maybe it's known bug and there is something I can do to avoid this one in the future?

- Neo4j version: 3.4.9
- Operating system: Ubuntu 18.04
- API/Driver: Cypher/cypher-shell

Here are few last logs:
```
Jan 23 15:28:05 neo4j[13029]: 2019-01-23 14:28:05.134+0000 ERROR Client triggered an unexpected error [Neo.DatabaseError.General.UnknownError]: head of empty list, reference 0e37ba47-6f4e-4b27-bbc7-555ca07cac29.
...
Jan 24 22:58:13 neo4j[13029]: 2019-01-24 21:58:13.731+0000 ERROR Client triggered an unexpected error [Neo.DatabaseError.General.UnknownError]: head of empty list, reference 8fbc6196-5d99-41e5-98d7-524da14cc84e.
...
2019-01-25 11:48:17.855+0000 ERROR [o.n.b.v.r.ErrorReporter] Client triggered an unexpected error [Neo.DatabaseError.General.UnknownError]: head of empty list, reference d0acce0f-1b7b-4eea-b01a-2dc1483082
57. head of empty list
java.util.NoSuchElementException: head of empty list
        at scala.collection.immutable.Nil$.head(List.scala:431)
        at scala.collection.immutable.Nil$.head(List.scala:428)
        at org.neo4j.cypher.internal.runtime.interpreted.commands.convert.CommunityExpressionConverter$.toCommandExpression(CommunityExpressionConverter.scala:204)
        at org.neo4j.cypher.internal.runtime.interpreted.commands.convert.CommunityExpressionConverter$.toCommandExpression(CommunityExpressionConverter.scala:70)
        at org.neo4j.cypher.internal.runtime.interpreted.commands.convert.ExpressionConverters$$anonfun$toCommandExpression$1.apply(ExpressionConverters.scala:43)
        at org.neo4j.cypher.internal.runtime.interpreted.commands.convert.ExpressionConverters$$anonfun$toCommandExpression$1.apply(ExpressionConverters.scala:42)
        at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
        at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:35)
        at org.neo4j.cypher.internal.runtime.interpreted.commands.convert.ExpressionConverters.toCommandExpression(ExpressionConverters.scala:42)
        at org.neo4j.cypher.internal.compatibility.v3_4.runtime.CommunityPipeBuilder$$anonfun$1.apply(CommunityPipeBuilder.scala:56)
        at org.neo4j.cypher.internal.compatibility.v3_4.runtime.CommunityPipeBuilder$$anonfun$1.apply(CommunityPipeBuilder.scala:56)
        at scala.Function1$$anonfun$andThen$1.apply(Function1.scala:52)
        at scala.Function1$$anonfun$andThen$1.apply(Function1.scala:52)
        at org.neo4j.cypher.internal.compatibility.v3_4.runtime.CommunityPipeBuilder$$anonfun$build$10.apply(CommunityPipeBuilder.scala:223)
        at org.neo4j.cypher.internal.compatibility.v3_4.runtime.CommunityPipeBuilder$$anonfun$build$10.apply(CommunityPipeBuilder.scala:223)
        at org.neo4j.cypher.internal.util.v3_4.Eagerly$$anonfun$mapToBuilder$1.apply(Eagerly.scala:41)
        at org.neo4j.cypher.internal.util.v3_4.Eagerly$$anonfun$mapToBuilder$1.apply(Eagerly.scala:41)
        at scala.collection.TraversableOnce$$anonfun$foldLeft$1.apply(TraversableOnce.scala:157)
        at scala.collection.TraversableOnce$$anonfun$foldLeft$1.apply(TraversableOnce.scala:157)
        at scala.collection.immutable.Map$Map1.foreach(Map.scala:116)
        at scala.collection.TraversableOnce$class.foldLeft(TraversableOnce.scala:157)
        at scala.collection.AbstractTraversable.foldLeft(Traversable.scala:104)
        at org.neo4j.cypher.internal.util.v3_4.Eagerly$.mapToBuilder(Eagerly.scala:41)
        at org.neo4j.cypher.internal.util.v3_4.Eagerly$.immutableMapValues(Eagerly.scala:34)
        at org.neo4j.cypher.internal.compatibility.v3_4.runtime.CommunityPipeBuilder.build(CommunityPipeBuilder.scala:223)
        at org.neo4j.cypher.internal.compatibility.v3_4.runtime.PipeExecutionPlanBuilder.org$neo4j$cypher$internal$compatibility$v3_4$runtime$PipeExecutionPlanBuilder$$buildPipe(PipeExecutionPlanBuilder.s
cala:117)
        at org.neo4j.cypher.internal.compatibility.v3_4.runtime.PipeExecutionPlanBuilder.build(PipeExecutionPlanBuilder.scala:42)
        at org.neo4j.cypher.internal.compatibility.v3_4.runtime.BuildInterpretedExecutionPlan$.process(BuildInterpretedExecutionPlan.scala:56)
        at org.neo4j.cypher.internal.compatibility.v3_4.runtime.BuildInterpretedExecutionPlan$.process(BuildInterpretedExecutionPlan.scala:40)
        at org.neo4j.cypher.internal.frontend.v3_4.phases.Phase$$anonfun$transform$1.apply(Phase.scala:36)
        at org.neo4j.cypher.internal.frontend.v3_4.helpers.package$$anonfun$closing$1.apply(package.scala:24)
        at org.neo4j.cypher.internal.frontend.v3_4.helpers.package$$anonfun$closing$1.apply(package.scala:24)
        at org.neo4j.cypher.internal.frontend.v3_4.helpers.package$.using(package.scala:32)
        at org.neo4j.cypher.internal.frontend.v3_4.helpers.package$.closing(package.scala:24)
        at org.neo4j.cypher.internal.frontend.v3_4.phases.Phase$class.transform(Phase.scala:35)
        at org.neo4j.cypher.internal.compatibility.v3_4.runtime.BuildInterpretedExecutionPlan$.transform(BuildInterpretedExecutionPlan.scala:40)
        at org.neo4j.cypher.internal.frontend.v3_4.phases.PipeLine.transform(Transformer.scala:58)
        at org.neo4j.cypher.internal.frontend.v3_4.phases.If.transform(Transformer.scala:87)
        at org.neo4j.cypher.internal.frontend.v3_4.phases.PipeLine.transform(Transformer.scala:62)
        at org.neo4j.cypher.internal.compatibility.v3_4.Compatibility$$anon$1$$anonfun$plan$1$$anon$2.produceWithExistingTX(Compatibility.scala:131)
        at org.neo4j.cypher.internal.compatibility.v3_4.Compatibility$$anon$1$$anonfun$plan$1$$anon$2.produceWithExistingTX(Compatibility.scala:124)
        at org.neo4j.cypher.internal.compatibility.QueryCache$$anonfun$1.apply(CacheAccessor.scala:41)
        at org.neo4j.cypher.internal.compatibility.MonitoringCacheAccessor$$anonfun$2.apply(CacheAccessor.scala:68)
        at org.neo4j.cypher.internal.compatibility.LFUCache$$anon$1.apply(LFUCache.scala:31)
        at com.github.benmanes.caffeine.cache.BoundedLocalCache.lambda$doComputeIfAbsent$14(BoundedLocalCache.java:2039)
        at java.util.concurrent.ConcurrentHashMap.compute(ConcurrentHashMap.java:1892)
        at com.github.benmanes.caffeine.cache.BoundedLocalCache.doComputeIfAbsent(BoundedLocalCache.java:2037)
        at com.github.benmanes.caffeine.cache.BoundedLocalCache.computeIfAbsent(BoundedLocalCache.java:2020)
        at com.github.benmanes.caffeine.cache.LocalCache.computeIfAbsent(LocalCache.java:112)
        at com.github.benmanes.caffeine.cache.LocalManualCache.get(LocalManualCache.java:54)
        at org.neo4j.cypher.internal.compatibility.LFUCache.getOrElseUpdate(LFUCache.scala:30)
        at org.neo4j.cypher.internal.compatibility.LFUCache.apply(LFUCache.scala:48)
        at org.neo4j.cypher.internal.compatibility.MonitoringCacheAccessor.getOrElseUpdate(CacheAccessor.scala:66)
        at org.neo4j.cypher.internal.compatibility.QueryCache.getOrElseUpdate(CacheAccessor.scala:39)
        at org.neo4j.cypher.internal.compatibility.v3_4.Compatibility$$anon$1$$anonfun$plan$1.apply(Compatibility.scala:136)
        at org.neo4j.cypher.internal.compatibility.v3_4.Compatibility$$anon$1$$anonfun$plan$1.apply(Compatibility.scala:104)
        at org.neo4j.cypher.exceptionHandler$runSafely$.apply(exceptionHandler.scala:89)
        at org.neo4j.cypher.internal.compatibility.v3_4.Compatibility$$anon$1.plan(Compatibility.scala:104)
        at org.neo4j.cypher.internal.ExecutionEngine$$anon$3.produceWithExistingTX(ExecutionEngine.scala:185)
        at org.neo4j.cypher.internal.ExecutionEngine$$anon$3.produceWithExistingTX(ExecutionEngine.scala:182)
        at org.neo4j.cypher.internal.compatibility.QueryCache$$anonfun$1.apply(CacheAccessor.scala:41)
        at org.neo4j.cypher.internal.compatibility.MonitoringCacheAccessor$$anonfun$2.apply(CacheAccessor.scala:68)
        at org.neo4j.cypher.internal.compatibility.LFUCache$$anon$1.apply(LFUCache.scala:31)
        at com.github.benmanes.caffeine.cache.BoundedLocalCache.lambda$doComputeIfAbsent$14(BoundedLocalCache.java:2039)
        at java.util.concurrent.ConcurrentHashMap.compute(ConcurrentHashMap.java:1853)
        at com.github.benmanes.caffeine.cache.BoundedLocalCache.doComputeIfAbsent(BoundedLocalCache.java:2037)
        at com.github.benmanes.caffeine.cache.BoundedLocalCache.computeIfAbsent(BoundedLocalCache.java:2020)
        at com.github.benmanes.caffeine.cache.LocalCache.computeIfAbsent(LocalCache.java:112)
        at com.github.benmanes.caffeine.cache.LocalManualCache.get(LocalManualCache.java:54)
        at org.neo4j.cypher.internal.compatibility.LFUCache.getOrElseUpdate(LFUCache.scala:30)
        at org.neo4j.cypher.internal.compatibility.LFUCache.apply(LFUCache.scala:48)
        at org.neo4j.cypher.internal.compatibility.MonitoringCacheAccessor.getOrElseUpdate(CacheAccessor.scala:66)
        at org.neo4j.cypher.internal.compatibility.QueryCache.getOrElseUpdate(CacheAccessor.scala:39)
        at org.neo4j.cypher.internal.ExecutionEngine.liftedTree1$1(ExecutionEngine.scala:190)
        at org.neo4j.cypher.internal.ExecutionEngine.planQuery(ExecutionEngine.scala:172)
        at org.neo4j.cypher.internal.ExecutionEngine.execute(ExecutionEngine.scala:115)
        at org.neo4j.cypher.internal.javacompat.ExecutionEngine.executeQuery(ExecutionEngine.java:61)
        at org.neo4j.bolt.v1.runtime.TransactionStateMachineSPI$1.start(TransactionStateMachineSPI.java:144)
        at org.neo4j.bolt.v1.runtime.TransactionStateMachine$State.startExecution(TransactionStateMachine.java:444)
        at org.neo4j.bolt.v1.runtime.TransactionStateMachine$State$1.execute(TransactionStateMachine.java:259)
        at org.neo4j.bolt.v1.runtime.TransactionStateMachine$State$1.run(TransactionStateMachine.java:240)
        at org.neo4j.bolt.v1.runtime.TransactionStateMachine.run(TransactionStateMachine.java:81)
        at org.neo4j.bolt.v1.runtime.BoltStateMachine$State$2.run(BoltStateMachine.java:457)
        at org.neo4j.bolt.v1.runtime.BoltStateMachine.run(BoltStateMachine.java:225)
        at org.neo4j.bolt.v1.messaging.BoltMessageRouter.lambda$onRun$3(BoltMessageRouter.java:93)
        at org.neo4j.bolt.runtime.DefaultBoltConnection.processNextBatch(DefaultBoltConnection.java:195)
        at org.neo4j.bolt.runtime.DefaultBoltConnection.processNextBatch(DefaultBoltConnection.java:143)
        at org.neo4j.bolt.runtime.ExecutorBoltScheduler.executeBatch(ExecutorBoltScheduler.java:170)
        at org.neo4j.bolt.runtime.ExecutorBoltScheduler.lambda$scheduleBatchOrHandleError$2(ExecutorBoltScheduler.java:153)
        at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1590)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
...
2019-01-28 10:22:20.574+0000 ERROR [o.n.b.v.r.ErrorReporter] Client triggered an unexpected error [Neo.DatabaseError.General.UnknownError]: head of empty list, reference 6578d77b-99f0-4b1a-98ba-c94d39f938
93. head of empty list
java.util.NoSuchElementException: head of empty list
        at scala.collection.immutable.Nil$.head(List.scala:431)
        at scala.collection.immutable.Nil$.head(List.scala:428)
        at org.neo4j.cypher.internal.runtime.interpreted.commands.convert.CommunityExpressionConverter$.toCommandExpression(CommunityExpressionConverter.scala:197)
        at org.neo4j.cypher.internal.runtime.interpreted.commands.convert.CommunityExpressionConverter$.toCommandExpression(CommunityExpressionConverter.scala:70)
        at org.neo4j.cypher.internal.runtime.interpreted.commands.convert.ExpressionConverters$$anonfun$toCommandExpression$1.apply(ExpressionConverters.scala:43)
        at org.neo4j.cypher.internal.runtime.interpreted.commands.convert.ExpressionConverters$$anonfun$toCommandExpression$1.apply(ExpressionConverters.scala:42)
        at scala.collection.IndexedSeqOptimized$class.foreach(IndexedSeqOptimized.scala:33)
        at scala.collection.mutable.WrappedArray.foreach(WrappedArray.scala:35)
        at org.neo4j.cypher.internal.runtime.interpreted.commands.convert.ExpressionConverters.toCommandExpression(ExpressionConverters.scala:42)
        at org.neo4j.cypher.internal.compatibility.v3_4.runtime.CommunityPipeBuilder$$anonfun$1.apply(CommunityPipeBuilder.scala:56)
        at org.neo4j.cypher.internal.compatibility.v3_4.runtime.CommunityPipeBuilder$$anonfun$1.apply(CommunityPipeBuilder.scala:56)
        at scala.Function1$$anonfun$andThen$1.apply(Function1.scala:52)
        at scala.Function1$$anonfun$andThen$1.apply(Function1.scala:52)
        at org.neo4j.cypher.internal.compatibility.v3_4.runtime.CommunityPipeBuilder$$anonfun$build$10.apply(CommunityPipeBuilder.scala:223)
        at org.neo4j.cypher.internal.compatibility.v3_4.runtime.CommunityPipeBuilder$$anonfun$build$10.apply(CommunityPipeBuilder.scala:223)
        at org.neo4j.cypher.internal.util.v3_4.Eagerly$$anonfun$mapToBuilder$1.apply(Eagerly.scala:41)
        at org.neo4j.cypher.internal.util.v3_4.Eagerly$$anonfun$mapToBuilder$1.apply(Eagerly.scala:41)
        at scala.collection.TraversableOnce$$anonfun$foldLeft$1.apply(TraversableOnce.scala:157)
        at scala.collection.TraversableOnce$$anonfun$foldLeft$1.apply(TraversableOnce.scala:157)
        at scala.collection.immutable.Map$Map1.foreach(Map.scala:116)
        at scala.collection.TraversableOnce$class.foldLeft(TraversableOnce.scala:157)
        at scala.collection.AbstractTraversable.foldLeft(Traversable.scala:104)
        at org.neo4j.cypher.internal.util.v3_4.Eagerly$.mapToBuilder(Eagerly.scala:41)
        at org.neo4j.cypher.internal.util.v3_4.Eagerly$.immutableMapValues(Eagerly.scala:34)
        at org.neo4j.cypher.internal.compatibility.v3_4.runtime.CommunityPipeBuilder.build(CommunityPipeBuilder.scala:223)
        at org.neo4j.cypher.internal.compatibility.v3_4.runtime.PipeExecutionPlanBuilder.org$neo4j$cypher$internal$compatibility$v3_4$runtime$PipeExecutionPlanBuilder$$buildPipe(PipeExecutionPlanBuilder.s
cala:117)
        at org.neo4j.cypher.internal.compatibility.v3_4.runtime.PipeExecutionPlanBuilder.build(PipeExecutionPlanBuilder.scala:42)
        at org.neo4j.cypher.internal.compatibility.v3_4.runtime.BuildInterpretedExecutionPlan$.process(BuildInterpretedExecutionPlan.scala:56)
        at org.neo4j.cypher.internal.compatibility.v3_4.runtime.BuildInterpretedExecutionPlan$.process(BuildInterpretedExecutionPlan.scala:40)
        at org.neo4j.cypher.internal.frontend.v3_4.phases.Phase$$anonfun$transform$1.apply(Phase.scala:36)
        at org.neo4j.cypher.internal.frontend.v3_4.helpers.package$$anonfun$closing$1.apply(package.scala:24)
        at org.neo4j.cypher.internal.frontend.v3_4.helpers.package$$anonfun$closing$1.apply(package.scala:24)
        at org.neo4j.cypher.internal.frontend.v3_4.helpers.package$.using(package.scala:32)
        at org.neo4j.cypher.internal.frontend.v3_4.helpers.package$.closing(package.scala:24)
        at org.neo4j.cypher.internal.frontend.v3_4.phases.Phase$class.transform(Phase.scala:35)
        at org.neo4j.cypher.internal.compatibility.v3_4.runtime.BuildInterpretedExecutionPlan$.transform(BuildInterpretedExecutionPlan.scala:40)
        at org.neo4j.cypher.internal.frontend.v3_4.phases.PipeLine.transform(Transformer.scala:58)
        at org.neo4j.cypher.internal.frontend.v3_4.phases.If.transform(Transformer.scala:87)
        at org.neo4j.cypher.internal.frontend.v3_4.phases.PipeLine.transform(Transformer.scala:62)
        at org.neo4j.cypher.internal.compatibility.v3_4.Compatibility$$anon$1$$anonfun$plan$1$$anon$2.produceWithExistingTX(Compatibility.scala:131)
        at org.neo4j.cypher.internal.compatibility.v3_4.Compatibility$$anon$1$$anonfun$plan$1$$anon$2.produceWithExistingTX(Compatibility.scala:124)
        at org.neo4j.cypher.internal.compatibility.QueryCache$$anonfun$1.apply(CacheAccessor.scala:41)
        at org.neo4j.cypher.internal.compatibility.MonitoringCacheAccessor$$anonfun$2.apply(CacheAccessor.scala:68)
        at org.neo4j.cypher.internal.compatibility.LFUCache$$anon$1.apply(LFUCache.scala:31)
        at com.github.benmanes.caffeine.cache.BoundedLocalCache.lambda$doComputeIfAbsent$14(BoundedLocalCache.java:2039)
        at java.util.concurrent.ConcurrentHashMap.compute(ConcurrentHashMap.java:1892)
        at com.github.benmanes.caffeine.cache.BoundedLocalCache.doComputeIfAbsent(BoundedLocalCache.java:2037)
        at com.github.benmanes.caffeine.cache.BoundedLocalCache.computeIfAbsent(BoundedLocalCache.java:2020)
        at com.github.benmanes.caffeine.cache.LocalCache.computeIfAbsent(LocalCache.java:112)
        at com.github.benmanes.caffeine.cache.LocalManualCache.get(LocalManualCache.java:54)
        at org.neo4j.cypher.internal.compatibility.LFUCache.getOrElseUpdate(LFUCache.scala:30)
        at org.neo4j.cypher.internal.compatibility.LFUCache.apply(LFUCache.scala:48)
        at org.neo4j.cypher.internal.compatibility.MonitoringCacheAccessor.getOrElseUpdate(CacheAccessor.scala:66)
        at org.neo4j.cypher.internal.compatibility.QueryCache.getOrElseUpdate(CacheAccessor.scala:39)
        at org.neo4j.cypher.internal.compatibility.v3_4.Compatibility$$anon$1$$anonfun$plan$1.apply(Compatibility.scala:136)
        at org.neo4j.cypher.internal.compatibility.v3_4.Compatibility$$anon$1$$anonfun$plan$1.apply(Compatibility.scala:104)
        at org.neo4j.cypher.exceptionHandler$runSafely$.apply(exceptionHandler.scala:89)
        at org.neo4j.cypher.internal.compatibility.v3_4.Compatibility$$anon$1.plan(Compatibility.scala:104)
        at org.neo4j.cypher.internal.ExecutionEngine$$anon$3.produceWithExistingTX(ExecutionEngine.scala:185)
        at org.neo4j.cypher.internal.ExecutionEngine$$anon$3.produceWithExistingTX(ExecutionEngine.scala:182)
        at org.neo4j.cypher.internal.compatibility.QueryCache$$anonfun$1.apply(CacheAccessor.scala:41)
        at org.neo4j.cypher.internal.compatibility.MonitoringCacheAccessor$$anonfun$2.apply(CacheAccessor.scala:68)
        at org.neo4j.cypher.internal.compatibility.LFUCache$$anon$1.apply(LFUCache.scala:31)
        at com.github.benmanes.caffeine.cache.BoundedLocalCache.lambda$doComputeIfAbsent$14(BoundedLocalCache.java:2039)
        at java.util.concurrent.ConcurrentHashMap.compute(ConcurrentHashMap.java:1892)
        at com.github.benmanes.caffeine.cache.BoundedLocalCache.doComputeIfAbsent(BoundedLocalCache.java:2037)
        at com.github.benmanes.caffeine.cache.BoundedLocalCache.computeIfAbsent(BoundedLocalCache.java:2020)
        at com.github.benmanes.caffeine.cache.LocalCache.computeIfAbsent(LocalCache.java:112)
        at com.github.benmanes.caffeine.cache.LocalManualCache.get(LocalManualCache.java:54)
        at org.neo4j.cypher.internal.compatibility.LFUCache.getOrElseUpdate(LFUCache.scala:30)
        at org.neo4j.cypher.internal.compatibility.LFUCache.apply(LFUCache.scala:48)
        at org.neo4j.cypher.internal.compatibility.MonitoringCacheAccessor.getOrElseUpdate(CacheAccessor.scala:66)
        at org.neo4j.cypher.internal.compatibility.QueryCache.getOrElseUpdate(CacheAccessor.scala:39)
        at org.neo4j.cypher.internal.ExecutionEngine.liftedTree1$1(ExecutionEngine.scala:190)
        at org.neo4j.cypher.internal.ExecutionEngine.planQuery(ExecutionEngine.scala:172)
        at org.neo4j.cypher.internal.ExecutionEngine.execute(ExecutionEngine.scala:115)
        at org.neo4j.cypher.internal.javacompat.ExecutionEngine.executeQuery(ExecutionEngine.java:61)
        at org.neo4j.bolt.v1.runtime.TransactionStateMachineSPI$1.start(TransactionStateMachineSPI.java:144)
        at org.neo4j.bolt.v1.runtime.TransactionStateMachine$State.startExecution(TransactionStateMachine.java:444)
        at org.neo4j.bolt.v1.runtime.TransactionStateMachine$State$1.execute(TransactionStateMachine.java:259)
        at org.neo4j.bolt.v1.runtime.TransactionStateMachine$State$1.run(TransactionStateMachine.java:240)
        at org.neo4j.bolt.v1.runtime.TransactionStateMachine.run(TransactionStateMachine.java:81)
        at org.neo4j.bolt.v1.runtime.BoltStateMachine$State$2.run(BoltStateMachine.java:457)
        at org.neo4j.bolt.v1.runtime.BoltStateMachine.run(BoltStateMachine.java:225)
        at org.neo4j.bolt.v1.messaging.BoltMessageRouter.lambda$onRun$3(BoltMessageRouter.java:93)
        at org.neo4j.bolt.runtime.DefaultBoltConnection.processNextBatch(DefaultBoltConnection.java:195)
        at org.neo4j.bolt.runtime.DefaultBoltConnection.processNextBatch(DefaultBoltConnection.java:143)
        at org.neo4j.bolt.runtime.ExecutorBoltScheduler.executeBatch(ExecutorBoltScheduler.java:170)
        at org.neo4j.bolt.runtime.ExecutorBoltScheduler.lambda$scheduleBatchOrHandleError$2(ExecutorBoltScheduler.java:153)
        at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1590)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

**Comment by chrisvest** [comment]
Do you see any stack trace if your search for that reference in the `debug.log`?

**Comment by gkostkowski** [comment]
Oh, I missed the fact that _journalctl_ gives incomplete log. I updated logs in the issue.

I attached two stack traces, rest is the same - error occurs in _CommunityExpressionConverter.scala:197_ or _CommunityExpressionConverter.scala:204_

**Comment by Hunterness** [comment]
Hi @gkostkowski, thanks for reporting, we will look into it and get back to you.

Best Regards,
Team Cypher

**Comment by sherfert** [comment]
This bug was fixed in 3.4.11

Updating to 3.4.11 or newer will resolve this issue. However, your queries will still fail, but get a nicer error message. The reason is that in some of your queries you are probably calling `min()` or `max()` without providing an argument to the function, which is forbidden.
